File: cash.c
#include <stdio.h> 
#include <cs50.h>
#include <math.h>

float get_positive_float(string prompt);
int quarters, dimes, nickels, pennies;

int main(void)
{
    float dollars = get_positive_float("Change owed: ");
    int cents = round(dollars * 100);
    for (quarters = 0; cents >= 25; quarters++)
    {
        cents -= 25; 
    }
    for (dimes = 0; cents >= 10; dimes++)
    {
        cents -= 10;
    }
    for (nickels = 0; cents >= 5; nickels++)
    {
        cents -= 5;
    }
    pennies = cents;
    printf("%i\n", quarters + dimes + nickels + cents);  
}


float get_positive_float(string prompt)
{
    float n;
    do
    {
      n = get_float("%s", prompt); 
    }
    while (n < 0);
    return n;
}

File: cash.py
# Problem Set 6, Sentimental-Cash, Python, 13 July 2019

from cs50 import get_float

# Get changed owed in dollars from user
while True:
    dollars = get_float("Changed owed: ")
    if dollars >= 0:
        break

# Convert dollars to cents
cents = round(dollars * 100)

coins = cents // 25
cents %= 25

coins = coins + cents // 10
cents %= 10

coins = coins + cents // 5
cents %= 5

print(coins + cents)